lib/contracts/code_templates/main-context.js.ejs

```diff
@@ -1 +1,5 @@
-var __mainContext = __mainContext || this;
+var __mainContext = __mainContext || (
+    typeof this !== 'undefined' ? this : (
```

Comment:
You can usually just do `this ? ...`

Reply:
Previously, the pipeline was relying on a behavior whereby webpack-packed modules would be in a function-scope where `this` was roughly the equivalent of `module.exports`. I say, roughly, because upon examining the old webpack output, there was clearly some other webpack magic happening.

In this revised pipeline, modules are getting packed as ES6 modules. In ES6, per speck, module scope is always in strict mode. And per spec, in strict mode `this` becomes `undefined`. Prior to the change you commented upon, when `var __mainContext = __mainContext || this` was transpiled it would be automatically changed to `var __mainContext = __mainContext || undefined`. I was puzzled why babel (or maybe it's webpack) would rewrite the source like that, so I researched and learned about the ES6 module strict mode rule.

The current revision ends up making `__mainContext` identify with `window`. That works okay, and `__embarkContext` is hanving off `window` anyway. The fallback to `void 0` is a sanity check in case we're running in an environment which the current initialization routines aren't designed to handle. That won't happen in a browser since `self === window`. The better thing to do is revise how we're bootstrapping the environment; more responsibility should be handed to EmbarkJS, so that dapps aren't so tightly wound to embark's code generation.

Reply:
You're right. I misunderstood what you were asking, and I was explaining why I altered the RHS of `||`. Change made, as you suggested.